### PR TITLE
[XPU][FP8] Add block-scaled GDN decode fusions

### PR DESCRIPTION
**Pull request by gc-fu**
## Summary

Add the custom ESIMD kernel support required by the next block-scaled FP8 GDN decode stage.

This PR adds:

- block-scaled FP8 GDN decode fusion kernels;
- C++/Python operator bindings and exports;
- focused block-scale fusion tests.

This is the kernel dependency only. The framework integration will be submitted separately after this PR is reviewed and merged.

## Base

- `intel/llm-scaler` main: `6e0ee089`
- Includes the previously merged fused block-MoE decode kernel.

## Validation plan

1. Build and install the custom ESIMD wheel from this branch.
2. Run the focused kernel tests.
3. After merge, create the framework GDN integration PR from source commit `b431470ff`.
4. Validate the combined stack with per-block GSM8K 30x100 and exact 32K input / 2K output performance at batch 1 and 4.

The combined framework validation will use XPU 0,1, oneCCL 2021.15, no `ONECCL_VARS` override, and `VLLM_XPU_ALLREDUCE_RETRY_ON_NAN=0`.

### CODE REVIEW

**Comment by gc-fu** [comment]
Joint validation with framework PR intel-sandbox/llm-scaler-vllm-xpu#142 completed on XPU 0,1.

Code under test:
- Kernel: `aec789f`
- Framework: `b7285ac0e`
- oneCCL: `/opt/intel/oneapi/ccl/2021.15/lib/libccl.so.1`

Kernel-focused test: PASS. All fused outputs had cosine similarity `1.0`; relative errors were between `1.595e-4` and `1.746e-4`. The dual GEMV microbenchmark measured `0.0376 ms` separate vs `0.0332 ms` fused.

GSM8K per-block accuracy: 24 x 100 questions, all rounds PASS, mean `98.625%`, range `97%-100%`, invalid `0`, NaN `0`, device errors `0`. The remaining 6 planned rounds were intentionally stopped to begin performance testing.

32k input / 2k output performance, same accuracy-warmed service:

| Batch | Output tok/s | TTFT (ms) | TPOT (ms) | E2E (ms) |
|---:|---:|---:|---:|---:|
| 1 | 37.9140 | 20163.833 | 16.538 | 54016.590 |
| 4 | 61.5222 | 52611.234 | 39.280 | 133018.311 |

All validation checks passed.

File: vllm/custom-esimd-kernels-vllm/tests/test_fp8_blockscale_fusions.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+"""Correctness and microbenchmark for block-scaled GDN decode fusions."""
+
+import statistics
+import time
+
+import torch
+
+import custom_esimd_kernels_vllm as esimd
+from test_fp8_blockscale_gemm import dequant_weight, quantize_weight_block
```
[the remark this answers is not in the document]
Comment (gc-fu) [reply vllm/custom-esimd-kernels-vllm/tests/test_fp8_blockscale_fusions.py:13]: Fixed in b2e138f. The test now adds its directory to sys.path for a reliable sibling import and exposes a pytest-collectable test function while preserving direct execution.

File: vllm/custom-esimd-kernels-vllm/python/custom_esimd_kernels_vllm/ops.py [hunk 2]
```diff
@@ -90,6 +90,28 @@ def esimd_gemv_fp8_pert_fused2(
     return _ops.esimd_gemv_fp8_pert_fused2(input, w0, s0, o0, w1, s1, o1)
 
 
+def esimd_gemv_fp8_blockscale_fused2(
+    input: torch.Tensor,
+    w0: torch.Tensor, s0: torch.Tensor, o0: torch.Tensor,
+    w1: torch.Tensor, s1: torch.Tensor, o1: torch.Tensor,
+    block_n: int = 128, block_k: int = 128,
+) -> torch.Tensor:
+    """Decode dual GEMV for two E4M3 weights with 128x128 block scales."""
```
[the remark this answers is not in the document]
Comment (gc-fu) [reply vllm/custom-esimd-kernels-vllm/python/custom_esimd_kernels_vllm/ops.py]: Fixed in b2e138f. Both fused2 wrappers now document that o0/o1 are mutated in place and that o0 is returned.

File: vllm/custom-esimd-kernels-vllm/csrc/xpu/esimd_kernel_gemm.sycl [hunk 3]
```diff
@@ -136,3 +137,115 @@ at::Tensor esimd_gemm_fp8_blockscale(
...
+                "esimd_norm_gemv_fp8_blockscale: weight must be fp8_e4m3");
+    TORCH_CHECK(gemv_scale.scalar_type() == at::ScalarType::Float,
+                "esimd_norm_gemv_fp8_blockscale: scale must be float32");
+    TORCH_CHECK(output.scalar_type() == at::ScalarType::Half,
+                "esimd_norm_gemv_fp8_blockscale: output must be fp16");
+    TORCH_CHECK(V == 128 && K == HV * V && K % 128 == 0,
+                "esimd_norm_gemv_fp8_blockscale: requires V=128 and K=HV*V");
+    TORCH_CHECK(gemv_scale.size(0) == (N + 127) / 128 &&
+                    gemv_scale.size(1) == K / 128,
+                "esimd_norm_gemv_fp8_blockscale: scale shape mismatch");
```
[the remark this answers is not in the document]
Comment (gc-fu) [reply vllm/custom-esimd-kernels-vllm/csrc/xpu/esimd_kernel_gemm.sycl:241]: Fixed in b2e138f. The entry point now validates x and z against HV*V and norm_weight against V before launching the kernel.
